arch/arm/core/aarch32/fault_s.S

```diff
@@ -78,14 +83,16 @@ SECTION_SUBSEC_FUNC(TEXT,__fault,z_arm_usage_fault)
 SECTION_SUBSEC_FUNC(TEXT,__fault,z_arm_secure_fault)
 #endif /* CONFIG_ARM_SECURE_FIRMWARE */
 SECTION_SUBSEC_FUNC(TEXT,__fault,z_arm_debug_monitor)
+#endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
+SECTION_SUBSEC_FUNC(TEXT,__fault,z_arm_exc_spurious)
 #elif defined(CONFIG_ARMV7_R)
```

Comment:
Since we are doing `#if defined(CONFIG_CPU_CORTEX_M)` above, I wonder if this should be changed to `#elif defined(CONFIG_CPU_CORTEX_R)` for consistency; after all, that should be the preferred approach since this is not ARM**v7**-specific.

Reply:
Yes, that's fine.
Done (also some more cleanup)